src/clojisr/v1/r.clj

```diff
@@ -124,39 +124,78 @@
      (r (format fmt n (name package)))
      (intern *ns* ns (r ns)))))
 
-(def r== (r "`==`"))
-(def r!= (r "`!=`"))
-(def r< (r "`<`"))
-(def r> (r "`>`"))
-(def r<= (r "`<=`"))
-(def r>= (r "`>=`"))
-(def r& (r "`&`"))
-(def r&& (r "`&&`"))
-(def r| (r "`||`"))
-(def r|| (r "`||`"))
-(def r! (r "`!`"))
-(def r$ (r "`$`"))
-
-(def captured-str
-  "For the R function [str](https://www.rdocumentation.org/packages/utils/versions/3.6.1/topics/str), we capture the standard output and return the corresponding string."
-  (r "function(x) capture.output(str(x))"))
-
-(def println-captured-str (comp println-r-lines captured-str))
-
-(def str-md (comp r-lines->md captured-str))
-
-(def r** (r "`^`"))
-(def rdiv (r "`/`"))
-(def r- (r "`-`"))
+(defn- intern-r-binary [clj-op op]
+  (intern *ns* (symbol clj-op)
+          (fn [e1 e2]
+            ((clojisr.v1.r/r (format  "`%s`" op)) e1 e2))))
+
+(run!
+ (fn [op] (intern-r-binary (str "r" op) op))
+ ["==", "!=" "<" ">" "<=" ">=" "&" "&&" "|" "||" 
+  ;"r!" 
+  "$" "-"])
+
+(intern *ns* (symbol "r!")
+        (fn [e]
+          ((clojisr.v1.r/r   "`!`") e)))
+
+
+;; (def r== (r "`==`"))
+;; (def r!= (r "`!=`"))
+;; (def r< (r "`<`"))
+;; (def r> (r "`>`"))
+;; (def r<= (r "`<=`"))
+;; (def r>= (r "`>=`"))
+;; (def r& (r "`&`"))
+;; (def r&& (r "`&&`"))
+;; (def r| (r "`||`"))
+;; (def r|| (r "`||`"))
+;; (def r! (r "`!`"))
+;; (def r$ (r "`$`"))
+
+
+
+
+ (defn- captured-str []
+   "For the R function [str](https://www.rdocumentation.org/packages/utils/versions/3.6.1/topics/str), we capture the standard output and return the corresponding string."
+   (r "function(x) capture.output(str(x))")  )
+
+ (defn println-captured-str[x] 
```

Comment:
Not sure what this did before, so could not test it.
How is it suppoed to be used ?

Reply:
I don't remember precisely. Some functions print something and it would be great to have the result as a string. But it's a guess (I need to remember this topic)

Reply:
for example `summary()` prints to stdout